File: 조성원/할인 행사.js
function solution(want, number, discount) {
	let answer = 0;
	const shoppingCart = want
		.reduce(
			(acc, item, index) => [...acc, ...new Array(number[index]).fill(item)],
			[],
		)
		.sort();

	for (let i = 0; i < discount.length; i++) {
		const membership = discount.slice(i, i + 10).sort();

		if (JSON.stringify(shoppingCart) === JSON.stringify(membership)) answer++;
	}

	return answer;
}

File: 김현경/할인_행사.py
from collections import Counter
def solution(want, number, discount):
    answer = 0
    want_dict = {}
    for i in range(len(want)):
        want_dict[want[i]] = number[i]

    for i in range(len(discount)-9):
        if want_dict == Counter(discount[i:i+10]):
            answer += 1
            
    return answer

# 처음 풀이
# def solution(want, number, discount):
#     answer = 0
#     want_dict = {}
    
#     for i in range(len(want)):
#         want_dict[want[i]] = number[i]

#     for i in range(len(discount) - 9):
#         tmp_dict = {}

#         for j in range(i, i + 10):
#             if discount[j] in tmp_dict:
#                 tmp_dict[discount[j]] += 1
#             else:
#                 tmp_dict[discount[j]] = 1

#         if want_dict == tmp_dict:
#             answer += 1
            
#     return answer


